lalrpop/src/grammar/parse_tree.rs

```diff
@@ -1052,7 +1109,13 @@ impl Display for TypeParameter {
 impl Display for TypeRef {
     fn fmt(&self, fmt: &mut Formatter<'_>) -> Result<(), Error> {
         match *self {
-            TypeRef::Tuple(ref types) => write!(fmt, "({})", Sep(", ", types)),
+            TypeRef::Tuple(ref types) => {
+                if types.len() == 1 {
+                    write!(fmt, "({}, )", types[0])
+                } else {
+                    write!(fmt, "({})", Sep(", ", types))
+                }
+            }
```

Comment:
Does this touch any of the code you changed? Or are you just changing it because it's talking about tuples? You don't change this part of the grammar